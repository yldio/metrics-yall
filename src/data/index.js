export default [
  {
    start: '2017-10-05 13:12:13',
    end: '2017-10-05 13:13:58',
    metrics: [
      {
        time: '2017-10-05 13:12:28',
        value: 97595392,
      },
      {
        time: '2017-10-05 13:12:43',
        value: 94142464,
      },
      {
        time: '2017-10-05 13:12:58',
        value: 63918080,
      },
      {
        time: '2017-10-05 13:13:13',
        value: 94023680,
      },
      {
        time: '2017-10-05 13:13:28',
        value: 99467264,
      },
      {
        time: '2017-10-05 13:13:43',
        value: 99737600,
      },
      {
        time: '2017-10-05 13:13:58',
        value: 86839296,
      }
    ],
  },
  {
    start: '2017-10-05 13:12:13',
    end: '2017-10-05 13:13:58',
    metrics: [
      {
        time: '2017-10-05 13:12:28',
        value: 99844096,
      },
      {
        time: '2017-10-05 13:12:43',
        value: 93655040,
      },
      {
        time: '2017-10-05 13:12:58',
        value: 64172032,
      },
      {
        time: '2017-10-05 13:13:13',
        value: 100237312,
      },
      {
        time: '2017-10-05 13:13:28',
        value: 99844096,
      },
      {
        time: '2017-10-05 13:13:43',
        value: 88035328,
      },
      {
        time: '2017-10-05 13:13:58',
        value: 100237312,
      }
    ],
  },
  {
    start: '2017-10-05 13:12:13',
    end: '2017-10-05 13:13:58',
    metrics: [
      {
        time: '2017-10-05 13:12:28',
        value: 98512896,
      },
      {
        time: '2017-10-05 13:12:43',
        value: 98258944,
      },
      {
        time: '2017-10-05 13:12:58',
        value: 91385856,
      },
      {
        time: '2017-10-05 13:13:13',
        value: 50900992,
      },
      {
        time: '2017-10-05 13:13:28',
        value: 98512896,
      },
      {
        time: '2017-10-05 13:13:43',
        value: 98033664,
      },
      {
        time: '2017-10-05 13:13:58',
        value: 80048128,
      }
    ],
  },
  {
    start: '2017-10-05 13:12:13',
    end: '2017-10-05 13:13:58',
    metrics: [
      {
        time: '2017-10-05 13:12:28',
        value: 95563776,
      },
      {
        time: '2017-10-05 13:12:43',
        value: 48996352,
      },
      {
        time: '2017-10-05 13:12:58',
        value: 89772032,
      },
      {
        time: '2017-10-05 13:13:13',
        value: 83755008,
      },
      {
        time: '2017-10-05 13:13:28',
        value: 101920768,
      },
      {
        time: '2017-10-05 13:13:43',
        value: 95576064,
      },
      {
        time: '2017-10-05 13:13:58',
        value: 101699584,
      }
    ],
  },
  {
    start: '2017-10-05 13:12:13',
    end: '2017-10-05 13:13:58',
    metrics: [
      {
        time: '2017-10-05 13:12:28',
        value: 101732352,
      },
      {
        time: '2017-10-05 13:12:43',
        value: 101732352,
      },
      {
        time: '2017-10-05 13:12:58',
        value: 94240768,
      },
      {
        time: '2017-10-05 13:13:13',
        value: 102371328,
      },
      {
        time: '2017-10-05 13:13:28',
        value: 61218816,
      },
      {
        time: '2017-10-05 13:13:43',
        value: 102371328,
      },
      {
        time: '2017-10-05 13:13:58',
        value: 100716544,
      }
    ],
  }
];
