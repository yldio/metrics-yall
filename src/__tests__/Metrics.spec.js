import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { render } from 'enzyme';
import data from '../data';
import Metrics from '../';

Enzyme.configure({ adapter: new Adapter() });

it('Renders <Metrics /> without throwing', () => {
  const tree = render(
    <Metrics
      metrics={data}
      graphDurationSeconds={90}
      width="100%"
      height="100%"
    />
  );
  expect(tree).toMatchSnapshot();
});

it('Renders <Metrics /> with title', () => {
  const tree = render(<Metrics metrics={data} title="stuff" />);
  expect(tree).toMatchSnapshot();
});

it('Renders <Metrics /> with legend', () => {
  const tree = render(<Metrics metrics={data} legend />);
  expect(tree).toMatchSnapshot();
});

it('Renders <Metrics /> with legend', () => {
  const tree = render(<Metrics metrics={data} tooltips />);
  expect(tree).toMatchSnapshot();
});

it('Renders <Metrics /> with chart colors', () => {
  const tree = render(
    <Metrics
      metrics={data}
      chartColors={[
        'rgb(255, 99, 132)',
        'rgb(255, 159, 64)',
        'rgb(255, 205, 86)',
        'rgb(75, 192, 192)',
        'rgb(54, 162, 235)',
        'rgb(153, 102, 255)',
        'rgb(201, 203, 207)'
      ]}
    />
  );
  expect(tree).toMatchSnapshot();
});

it('Renders <Metrics /> with unit', () => {
  const tree = render(
    <Metrics
      metrics={data}
      unit="hours"
    />
  );
  expect(tree).toMatchSnapshot();
});
