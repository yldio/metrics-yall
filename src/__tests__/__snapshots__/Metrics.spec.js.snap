// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Metrics /> with chart colors 1`] = `
<div
  style="width:500px;height:500px"
>
  <canvas />
</div>
`;

exports[`Renders <Metrics /> with legend 1`] = `
<div
  style="width:500px;height:500px"
>
  <canvas />
</div>
`;

exports[`Renders <Metrics /> with legend 2`] = `
<div
  style="width:500px;height:500px"
>
  <canvas />
</div>
`;

exports[`Renders <Metrics /> with title 1`] = `
<div
  style="width:500px;height:500px"
>
  <canvas />
</div>
`;

exports[`Renders <Metrics /> with unit 1`] = `
<div
  style="width:500px;height:500px"
>
  <canvas />
</div>
`;

exports[`Renders <Metrics /> without throwing 1`] = `
<div
  style="width:100%;height:100%"
>
  <canvas />
</div>
`;
